python/cuml/cuml/internals/api_decorators.py

```diff
@@ -55,27 +55,32 @@ def _find_arg(sig, arg_name, default_position):
 
     # Check for default name in input args
     if arg_name in sig.parameters:
-        return arg_name, params.index(arg_name)
+        param = sig.parameters[arg_name]
```

Comment:
Can't remember off the top of my head, but the refactor of decorators you are doing removes this auto inspection of signature, no? Just wondering since it'd be nice to not have magic behavior of this.

Reply:
Can you explain a bit more? The code here looks more or less the same, so maybe you mean something that effects something somewhere else because of this change?

Reply:
@dantegd I am a big confused by your question as well. Yes, the infrastructure revision will eliminate much of the magic and guess work, but here I am just inspecting the signature similar like before, but account for a provided default argument similar to how @betatim structured this in the earlier PR.